### ISSUE TITLE
please backport fix for GD3DX7 Z-buffer destruction crash

### ISSUE THREAD

**Issue by stefson** [kirides/GD3D11#93]
there is an unhandled exception error in gothic1 with 17.8-dev1-rl , which is triggered by returning to desktop from the menu. 
For some users it also happens when loading a save game

this is the error: 

```
======================================= UNHANDLED EXCEPTION OCCURED ======================================================
======================================= CRASH INFOS: =====================================================================
GOTHIC 1.08k_mod, Parser Version: 50
User:  tim,  CPUType: 586,  Mem: 2048 MB total, 2048 MB free
Startup Options:-game:dunkle_geheimnisse.ini
=============================================== CALLSTACK : ==============================================================
0023:00000000 (0x0BA98BEC 0x68A68ED0 0x00000000 0x0BA98B24) GothicMod.exe
0023:023794B9 (0x0BA98B24 0x1471C0E8 0x0126F8F4 0x0075FA11) g1_avx.dll
0023:0237A998 (0x0BA98B24 0x1471C168 0x1471C0E8 0x00000000) g1_avx.dll
0023:0075FA11 (0x00000001 0x00000000 0x1471C0E8 0x00000000) GothicMod.exe, CD3duContext::_DestroyDXBuffers()+174 byte(s)
0023:0075FCBC (0x01ED91DC 0x0075E82D 0x00000000 0x0075C98B) GothicMod.exe, CD3duContext::~CD3duContext()+182 byte(s)
0023:0075B702 (0x00000000 0x0075C98B 0x01ED91DC 0x0075E9A6) GothicMod.exe, CD3duContext::`scalar deleting destructor'()+8 byte(s)

0023:0075C98B (0x00000001 0x00000000 0x0075ECFD 0x1471C0E8) GothicMod.exe, CD3duContextNode::`scalar deleting destructor'()+8 byte(s)
0023:0075E9A6 (0x1471C0E8 0x007D06E8 0x00712828 0x1471C0E8) GothicMod.exe, CD3duContextList::DestroyD3duContext()+38 byte(s)
0023:0075ECFD (0x1471C0E8 0x7FD50034 0x00000000 0x0126FA48) GothicMod.exe, CD3duContext::Release()+29 byte(s)
0023:00712828 (0x00120000 0x00000000 0x7FD50034 0x7BC26BEE) GothicMod.exe, zCRnd_D3D::XD3D_ClearDevice()+1080 byte(s), D:\dev\gothic\current_work\ZenGin\_Kurt\zRndD3D_Init.cpp, line 1168
0023:7BC26BEE (0x00120094 0x01ED00B0 0x0BA98FD4 0x14664F34) ntdll.dll, RtlIsValidIndexHandle()+8318 byte(s)
0023:7BC26BEE (0x00000000 0x00000690 0x0000041A 0x00000020) ntdll.dll, RtlIsValidIndexHandle()+8318 byte(s)
0023:00000000 (0x00717150 0x00716F50 0x00714B60 0x00716860) GothicMod.exe
```

the fix is here: https://github.com/SaiyansKing/GD3D11/commit/ae762dc5c743d3314e29afd1feac9a07e418c329
I compiled and tested it with your fork, plus this single commit patched on top of it, that fixed the error for me. 

thank you

**Comment by kirides** [comment]
already included as of [v17.8-dev4-ci](https://github.com/kirides/GD3D11/releases/tag/v17.8-dev4-ci)

**Comment by stefson** [comment]
ah, thanks for letting me know
